Add a minimal template.

Contrepoison opened the pull request:
Adding a minimal template with no GUI. Some config files were also removed.

Obarun commented:
ok, so a few remarks : 

into customizeChroot file : 
  
maybe you can remove the creation of the supervision tree in the config_s6 function, or only remove the dbus service activation. You can remove dbus_launch function too. This is a minimal template, dbus is not needed.

concerning file for package list : 

You can remove empty file like X_server, desktop.
Into services file, you can remove dbus-{s6rcserv,s6serv}, maybe rtkit too.
Into utilities files, rtkit, udisks2,upower,xdg-user-dirs,dbus,colord,libatasmart,polkit can be removed. Removing elinks, for me, it's not a good idea because we need browser on console to debug installation.
NOTE : You can remove, rename as you want all the package list file execpt for the base file which must named base and must contain the package name for base installation. this file is the first file installed by obarun-install script. You can use group name as package name into those files.

The /etc/X11 can be removed entirely, we don't have X server.
The etc/skel/Desktop directory can be removed. the /etc/skel/.config directory can be removed too.
The /etc/skel/.zlogin file can be removed, again no X. Keep .xinitrc can be good to provide a minimal example, i don't know.

The /usr directory can be removed entirely. this is concern X configuration too.

Contrepoison commented:
> remove the creation of the supervision tree in the config_s6 function, or only remove the dbus service activation. You can remove dbus_launch function too. This is a minimal template, dbus is not needed.

That's right.
I commented the part about the supervision tree in the config_s6 function, as well as the dbus_launch function.

>Into services file ...
>Into utilities files ...

Removed.

>Removing elinks, for me, it's not a good idea because we need browser on console to debug installation.

I'm not quite sure about this point. 
- If the internet does not work, elinks will not help. 
- If the internet connexion work and the installation is fine, the user can install elinks if needed.
- If the internet connexion work but the installation is broken, the user can install elinks.

Furthermore the user can prefer another console browser... I added some console browser (commented) in the utilities files.

>Keep .xinitrc can be good to provide a minimal example, i don't know.

Agree with that. I applied the rest of your recommendations.
